Add a note about active-repositories (#60)
input-output-hk/cardano-haskell-packages: README.md

index-state: cardano-haskell-packages 2022-08-25T00:00:00Z
```

> You _can_ use the `active-repositories` stanza if you like.

Reviewer: ```suggestion
If you want to state explicitly the repositories that should be considered for your project,  you can use the `active-repositories` stanza.
```

This explicitly tells cabal which repositories to use and with which precedence.
You don't need to set this, because cabal implicitly does the right thing (CHaP will take 
precedence, since it is defined in the project file).

```
-- note: later = higher precedence
active-repositories: hackage.haskell.org, cardano-haskell-packages
```

IMPORTANT: do _not_ use the `:override` modifier on CHaP. 

> This is in principle a good idea, but since we add patched Hackage packages to CHaP, it 

Reviewer: ```suggestion
The `:override` modifier tells cabal that CHaP is the sole source for any package it has available; while the default behaviour would be to consider, for each package, all the versions across all active repositories.
While this might sounds like a good idea, this
```

> would prevent cabal from taking later versions of those packages from Hackage, which is not what we want.

Reviewer: ```suggestion
would prevent cabal from taking later versions of those packages we patched from Hackage.
```
